src/auth/src/token_cache.rs

```diff
@@ -149,10 +157,18 @@ async fn refresh_task<T>(
                     break;
                 }
             }
-            Err(_) => {
-                // The retry policy has been used already by the inner token provider.
-                // If it ended in an error, just quit the background task.
-                break;
+            Err(e) => {
```

Comment:
Comment for ln 132. Lets assume there is a token in tx currently that is valid for 5 more mins, do we update the token in the tx with a error even though we are going to retry and may get a valid token within next 5 mins?

(Previously also we replaced valid token in tx with and error but that was ok since we do not retry and wont recover anyway)